mitosheet/src/components/taskpanes/SplitTextToColumns/SplitTextToColumnsTaskpane.tsx

```diff
@@ -0,0 +1,315 @@
+import React, { useEffect, useState } from "react";
+import MitoAPI from "../../../jupyter/api";
+import { AnalysisData, ColumnID, SheetData, SplitTextToColumnsParams, StepType, UIState } from "../../../types"
+import DefaultEmptyTaskpane from "../DefaultTaskpane/DefaultEmptyTaskpane";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import Row from "../../spacing/Row";
+import Col from "../../spacing/Col";
+import Select from "../../elements/Select";
+import DropdownItem from "../../elements/DropdownItem";
+import { getDisplayColumnHeader, getNewColumnHeader } from "../../../utils/columnHeaders";
+import useSendEditOnClick from "../../../hooks/useSendEditOnClick";
+import TextButton from "../../elements/TextButton";
+import Input from "../../elements/Input";
+import '../../../../css/taskpanes/SplitTextToColumns/SplitTextToColumns.css'
+import DefaultTaskpaneFooter from "../DefaultTaskpane/DefaultTaskpaneFooter";
+import MultiSelectButtonItem from "../../elements/MulitSelectButtonItem";
+import { toggleInArray } from "../../../utils/arrays";
+import Spacer from "../../spacing/Spacer";
+
+interface SplitTextToColumnsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    dfNames: string[];
+    startingColumnID: ColumnID | undefined
+}
+
+interface SplitTextToColumnsResult {
+    num_cols_created: number;
+}
+
+const delimiters = {',': 'Comma', '-': 'Dash', '\t': 'Tab', ' ': 'Space'}
+
+const getDefaultParams  = (startingColumnID: ColumnID | undefined, sheetDataArray: SheetData[], sheetIndex: number): SplitTextToColumnsParams | undefined => {
+    if (sheetDataArray.length === 0 || sheetDataArray[sheetIndex] === undefined && sheetIndex > 0) {
```

Comment:
you need to add some parens on this statement, its' very risky as is. The second clause after the or should always be logically grouped (right now you're relying on JS execution order that is fragile).